source/pythonConsole.py

```diff
@@ -335,6 +358,55 @@ def onInputChar(self, evt):
 			self.Close()
 			return
 		evt.Skip()
+	
+	def onInputPaste(self, evt):
+		# Test if text content is available
+		if not wx.TheClipboard.IsSupported(wx.DataFormat(wx.DF_TEXT)):
+			evt.Skip()
+			return
+		cpTextData = wx.TextDataObject()
+		if not wx.TheClipboard.Open():
+			evt.Skip()
+			return
+		try:
+			res = wx.TheClipboard.GetData(cpTextData)
+		except:
+			log.exception()
+			evt.Skip()
+			return
+		finally:
+			wx.TheClipboard.Close()
+		if not res:
+			# Retrieval of text data failed
+			evt.Skip()
+			return
+		cpText = cpTextData.GetText()
+		if not cpText.strip():
+			evt.Skip()
+			return
+		cpLines = cpText.split("\n")

```

Comment:
I agree that \n is the safest split point here. `\r\n` would fail for unix style clipboard data.